geps/gep-3792/index.md

```diff
@@ -246,16 +233,254 @@ guarantee that **all four** of the problems listed above need resolutions, and
 must have implementation from at least two different Gateways and two
 different meshes.
 
-### Gateway for Ingress (North/South)
+## API
+
+There are three important aspects to the OCG API:
+
+1. The API must allow the OCG to be configured to securely communicate with
+   meshed workloads in the cluster, including providing the OCG with the
+   information it needs to authenticate itself as a workload.
+
+2. The API must allow the mesh to be configured to accept requests from the
+   OCG, including providing the mesh with the information it needs to
+   authenticate the OCG as a workload.
+
+3. Since the API should affect only Gateway API resources, it is not a good
+   candidate for policy attachment. It is likely to be much more reasonable to
+   simply provide whatever extra configuration we need inline in the Gateway
+   or Mesh resources.
+
+The API must also solve all four of the problems listed above, so we'll start
+with an overview of the solutions before diving into the API details.
+
+### Solving the Trust Problem
+
+The trust problem is that the OCG and the mesh need to be able to trust each
+other. The simplest solution to this problem is to add a _trust bundle_ to the
+Gateway resource, and to create a Mesh resource which will also have a trust
+bundle:
+
+- The trust bundle in the Gateway resource will define the CA certificate(s)
```

Comment:
~~A single trust bundle implies that multitenancy (i.e. multiple meshes with different trust domains served by the same gateway) is a non-goal for now?~~

I see below that it's a list of secrets; should this be pluralized?

Reply:
"Trust bundle" to mean "one or more CA certificates that you're going to trust" is pretty well-established in the PKI world. Pluralizing it would imply having multiple lists, rather than multiple certificates; I don't really think that that makes sense.

Reply:
Oh wait I was interpreting this in light of John's comment about using CTB. In which case it got sort of confusing

Reply:
To respond to your overstruck comment: you are correct, support for multiple meshes in the same cluster is a non-goal in this pass. I'll make that explicit as well.